backend/app/controllers/spree/admin/locale_controller.rb

```diff
@@ -4,21 +4,22 @@ module Spree
   module Admin
     class LocaleController < Spree::Admin::BaseController
       def set
-        locale = params[:switch_to_locale].to_s.presence
+        requested_locale = params[:switch_to_locale].to_s.presence
 
-        if locale && I18n.available_locales.include?(locale.to_sym)
-          I18n.locale = locale
-          session[set_user_language_locale_key] = locale
-
-          respond_to do |format|
-            format.json { render json: { locale: locale, location: spree.admin_url } }
-          end
+        if locale_is_available?(requested_locale)
+          I18n.locale = requested_locale
+          session[set_user_language_locale_key] = requested_locale
+          render json: { locale: requested_locale, location: spree.admin_url }
         else
-          respond_to do |format|
-            format.json { render json: { locale: I18n.locale }, status: 404 }
-          end
+          render json: { locale: I18n.locale }, status: 404
```

Comment:
I'd gradually refactor the codebase following the linter rules when some lines are changed. This way we keep improving the codebase without big refactors that can compromise git history. The trade-off here is not being consistent, which I don't like either. I'd go with the first, if we have a linter setup with Hound (only checks lines changed) we should use it otherwise it doesn't make sense to keep it.

Reply:
Great to know for the future! Thank you 👍